This patch adds autodetection support for the Poulsbo ("psb") driver
needed by moblin.org for the Intel mobile chipset.  Rather than add an
else if, I restructured things with a case statement to make it conform
in style to the savage driver section.

Bryce Harrington  23-Aug-2007

Index: xorg-server/hw/xfree86/common/xf86AutoConfig.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xf86AutoConfig.c	2008-09-02 10:47:16.000000000 +0300
+++ xorg-server/hw/xfree86/common/xf86AutoConfig.c	2008-09-02 10:58:57.000000000 +0300
@@ -166,11 +166,16 @@
 	case 0x102c:		    driverList[0] = "chips"; break;
 	case 0x1013:		    driverList[0] = "cirrus"; break;
 	case 0x8086:
-	    if ((dev->device_id == 0x00d1) || (dev->device_id == 0x7800)) {
-		driverList[0] = "i740";
-	    } else {
-		driverList[0] = "intel";
-		driverList[1] = "i810";
+	    switch (dev->device_id)
+	    {
+		case 0x00d1: case 0x7800:
+		    driverList[0] = "i740"; break;
+		case 0x8108:
+		    driverList[0] = "psb"; break;
+		default:
+		    driverList[0] = "intel";
+		    driverList[1] = "i810";
+		    break;
 	    }
 	    break;
 	case 0x102b:		    driverList[0] = "mga";	break;
